cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -1342,6 +1366,27 @@ TEST_F(TestUInt32ParquetIO, Parquet_1_0_Compatibility) {
 
 using TestStringParquetIO = TestParquetIO<::arrow::StringType>;
 
+TEST_F(TestStringParquetIO, NonOverflowStringWithUseLargeBinaryVariantsSetting) {
+  std::shared_ptr<Array> values;
+
+  ::arrow::StringBuilder builder;
+  for (size_t i = 0; i < SMALL_SIZE; i++) {
+    ASSERT_OK(builder.Append("abc"));
+  }
+  ASSERT_OK(builder.Finish(&values));
```

Comment:
Many tests in this file probably use this builder-based pattern, but nowadays we have easier ways of building sample arrays: you can use `ArrayFromJSON`.
For example `ArrayFromJSON(utf8(), R("["abc", "def", null]"))`.